### DISCUSSION TITLE
runTasks_ failing - advice for troubleshooting

### DISCUSSION THREAD

**Discussion by ladywhiskers** [Chimildic/goofy#173] [Помощь с ошибкой] [answered]
Since updating my library code, runTasks_ has failed every execution, and all my scripts are failing as well. From what I can see, too many requests are being sent to Spotify and things are timing out.

What steps can I follow to troubleshoot this? I have reset Authorisation and turned off other triggers.  I haven't made major changes to codes either, and all were working last week.

![image](https://user-images.githubusercontent.com/99696830/154861003-dc69406c-c7ca-400c-9c66-2c9e1af5c8f4.png)

Here are two executions of a playlist code, one before the error started happening, and one after. No other changes were made to the code but updating the library.
![image](https://user-images.githubusercontent.com/99696830/154861127-e430acd6-584a-455b-8d5b-5c95563d2ddb.png)

**Comment by Chimildic** [comment, marked as answer]
1. How many saved tracks, artists and albums do you have? Do you try to get their in scripts?
2. What triggers do you set? (count and time)

**Reply by ladywhiskers** [reply]
Hi, I have around 4000 saved tracks, 40 saved artists, and 40 saved albums. 
Yesterday I changed a whole bunch of the scripts to `let savedTracks = Cache.read('SavedTracks');` instead of `let savedTracks = Source.getSavedTracks.

Not including runTasks I have 13 triggers, 2 update daily, and the rest update once a week.

This morning it looks like some of the scripts were running okay. I had a look at the Spotify Dashboard and there is a huge increase in /v1/search endpoints from 17 February, and I am not sure why. I have been using the scripts much the same since 7 February.
![image](https://user-images.githubusercontent.com/99696830/155017013-a9a4b9e0-33d9-4b60-9f37-30f7fca961df.png)

**Reply by Chimildic** [reply]
It's very strange. Can you remember what did you do on 16 February? We need to find source code that send search requests. 
Probably, do you use lastfm functions or fm radio?

The Spotify App (client and secret keys) uses only for goofy project?

**Reply by ladywhiskers** [reply]
It is strange! Every script seems to be running as normal now. I'd still like to find what is sending the search requests for my own education.

I have last.fm set up, but I don't utilise any of the last.fm features in my regular scripts.
I only use the Spotify client and secret keys with goofy.

I think I added the below code around 16 February (and forgot about it) - would this be causing the problem?
``` js
let recomSummerTracks = Source.getRecomTracks({
    seed_genres: 'summer',
  })
```

If I change this to have a limit of recommended tracks, such as 10, will that limit the sending of search requests?

**Reply by Chimildic** [reply]
Maybe did you share client and secret keys with somebody? You mustn't do this. 

Is use `mineTracks` or `getReleasesByArtists`? 

The main region of search requests is `Lastfm` module. Default config with lastfm params can't send so MANY requests. Only if use lastfm functions directly.

> would this be causing the problem?

No, `getRecomTracks` hasn't search requests. 

> Every script seems to be running as normal now

Is search requests zero now?

**Reply by ladywhiskers** [reply]
I definitely didn't share keys, unless someone else logged into my account.
No use of `mineTracks` or `getReleasesByArtists`, but I have checked my logs again and I did test out Lastfm.rangeTags on one of my playlists - so I suppose this is the cause.

Looks like search is back down to zero - graph is below. I'm curious why it was so high still on 18,19 and 20 when everything failed, but its back to normal now so I'll mark as resolved. Thank you for your help!
![image](https://user-images.githubusercontent.com/99696830/155294052-443ccd1c-3803-4c19-9a17-2d729cfdd5b4.png)

**Reply by Chimildic** [reply]
`rangeTags` send requests to lastfm, no spotify, no search request. 
I still don't know why so many requests were sent.

**Reply by ladywhiskers** [reply]
So strange! I'll keep monitoring and report if it happens again. I'm assuming that I messed up some code somewhere that caused it and have since fixed and forgotten what it was 🤷‍♀️
